docs/core/deploying/trimming/trimming-options.md

```diff
@@ -20,12 +21,44 @@ Trimming with `PublishTrimmed` was introduced in .NET Core 3.0. The other option
 
 Place this setting in the project file to ensure that the setting applies during `dotnet build`, not just `dotnet publish`.
 
+:::zone pivot="dotnet-7-0"
+
+This setting enables trimming and will trim all assemblies by default. In .NET 6, only assemblies which opted-in
+to trimming via `[AssemblyMetadata("IsTrimmable", "True")]` would be trimmed by default. You can return to the
```

Comment:
Good suggestion, but I'm actually not quite sure what we want to do here.

Do we want to define trimmable as anything marked `IsTrimmable=true`? Or do we want to say that it's any assembly that's trimmed?

If it's the former, then `TrimMode=full` trims non-trimmable assemblies. If it's the latter, then there's no direct mapping between `IsTrimmable` and "trimmable".

Reply:
I'd say something like "`TrimMode=full` trims all assemblies unless they're explicitly marked as trim-rooted assemblies" and define "trim-rooted" (or whatever term we want).

Reply:
The suggestion is to define a term that we can use to describe assemblies that are trimmed in TrimMode=partial (or assemblies to which TrimMode applies before .NET 7).

Basically start with saying we have two classes of assemblies and there are controls for what happens to each of the class. And then we have an exception mechanism to treat assemblies from one class as another class.

Once we have the term, we can say:

* < .NET 7 assemblies that are manifested as trimmable get trimmed
* => .NET 7 all assemblies are trimmed, TrimMode=partial is how to get the old behavior

I think it can simplify other parts of the doc too.

Reply:
By this definition, in .NET 7 we're trimming "non-trimmable" assemblies. I'm not sure "trimmable" is a good term to use here because I don't think it makes sense without a lot of context.

Reply:
We cemented ourselves into that term by calling it `IsTrimmable` everywhere (item metadata, the attribute). In .NET 6 we would trim assemblies that are not "trimmable assemblies" if we set TrimmerDefaultAction=link. We could call it differently (trimmable-manifested assemblies?). Or just embrace it...

Reply:
Or try going with "always-eligible-for-trimming"? (Even that one is not precise if someone goes and sets TrimMode=copy in .NET 6.)

I would probably really just go with "trimmable assembly" and make it obvious that it's not the assemblies that will be trimmed, it's just a name of a category that can be handled differently.